src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/Hdced4Calculator.kt

```diff
@@ -62,23 +62,45 @@ class Hdced4Calculator(val hdcedConfiguration: Hdced4Configuration) {
-      log.info("Quarter sentence length: {}", quarterSentenceLength)
-
-      sentenceCalculation.homeDetentionCurfew4PlusEligibilityDate = sdsPlusNotionalCrd.plusDays(quarterSentenceLength).minusDays(sentence.sentenceCalculation.calculatedTotalDeductedDays.toLong())
+    } else if (sentence is ConsecutiveSentence && !sentence.isSdsPlus()) {
+      if (sentence.orderedSentences.any { it.isSdsPlus() }) {
+        val sdsPlusSentences = sentence.orderedSentences.filter { it.isSdsPlus() }
+        val nonSdsPluSentences = sentence.orderedSentences.filter { !it.isSdsPlus() }
+        val sdsPlusLengthInDays = sdsPlusSentences.sumOf { it.getLengthInDays() }.toLong()
+        val nonSdsPlusSentenceLengthInDays = nonSdsPluSentences.sumOf { it.getLengthInDays() }.toLong()
+        log.info("Total sds plus length in days sentences: {}", sdsPlusLengthInDays)
+        val sdsPlusNotionalSled = sentence.sentencedAt.plusDays(sdsPlusLengthInDays)
+        log.info("Sds plus notional sled: {}", sdsPlusNotionalSled)
+        val sdsPlusNotionalCrd = sentence.sentencedAt.minusDays(1).plusDays(ceil(sdsPlusLengthInDays * (2.0 / 3.0)).toLong())
+        log.info("Sds plus notional crd: {}", sdsPlusNotionalCrd)
+        val lengthInDaysOfNonSdsPlusSentences = sentence.orderedSentences.filter { !it.isSdsPlus() }.sumOf { it.totalDuration().getLengthInDays(sdsPlusNotionalCrd) }.toLong()
+        log.info("Length in days of non sds plus sentences: {}", lengthInDaysOfNonSdsPlusSentences)
+
+        val sdsNotionalSled = sdsPlusNotionalCrd.plusDays(lengthInDaysOfNonSdsPlusSentences)
+        log.info("Sds notional sled: {}", sdsNotionalSled)
+        if (nonSdsPlusSentenceLengthInDays < hdcedConfiguration.envelopeMidPoint) {
+          val quarterSentenceLengthOr28Days = if (ceil(lengthInDaysOfNonSdsPlusSentences / 4.0).toLong() > 28) ceil(lengthInDaysOfNonSdsPlusSentences / 4.0).toLong() else 28
+          log.info("Quarter sentence length: {}", quarterSentenceLengthOr28Days)
+          sentenceCalculation.homeDetentionCurfew4PlusEligibilityDate = sdsPlusNotionalCrd.plusDays(quarterSentenceLengthOr28Days).minusDays(sentence.sentenceCalculation.calculatedTotalDeductedDays.toLong()).plusDays(sentence.sentenceCalculation.calculatedTotalAddedDays.toLong())
+        } else {
+          sentenceCalculation.homeDetentionCurfew4PlusEligibilityDate = sdsPlusNotionalCrd.minusDays(hdcedConfiguration.deductionDays)
+        }
+      } else {
+        val nonSdsPluSentences = sentence.orderedSentences.filter { !it.isSdsPlus() }
+        val nonSdsPlusSentenceLengthInDays = nonSdsPluSentences.sumOf { it.getLengthInDays() }.toLong()
+
+        if (nonSdsPlusSentenceLengthInDays < hdcedConfiguration.envelopeMidPoint) {
+          val quarterSentenceLengthOr28Days = if (ceil(sentenceCalculation.numberOfDaysToSentenceExpiryDate / 4.0).toLong() > 28) ceil(sentenceCalculation.numberOfDaysToSentenceExpiryDate / 4.0).toLong() else 28
```

Comment:
Done differently here to further down in  `calculateHdcedUnderMidpoint`

Reply:
Done.